### PR TITLE
Feature/77 - Dashboard logic pt.2

### LINKED ISSUE #77: Logic for dashboard
**As a** user on the dashboard page **I want** to see an overview of my habits, add, delete and edit habits.

With the logic for the dashboard implemented, the following should be possible:

- Display the user's habits, sorted by when they are next due
- Display the correct details for the habit
- Log a habit
- Filter habits by tag, difficult and/or type
- View habits that are due
- View habits that are upcoming
- View a list of all logged habits

Also to include (this was not in the design, although it is important 😅) is an option to add, edit and delete habits. Adding and editing can pass the habit along to the add/edit habit form.

### PR DESCRIPTION
**Pull request by Puritanic**
Habit due dates and Dashboard filters. 

**Link to Issue**:

-   Related to #77 and #99 

**Readiness**:

-   [x] Avoid duplicated logic
-   [ ] Unit test component functionalities
-   [x] Use semantic HTML tags
-   [x] Use [semantic commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
-   [x] Connect this Pull Request to an issue

### CODE REVIEW

File: client/src/components/Dashboard/Dashboard.styled.js [hunk 1]
```diff
@@ -35,6 +36,8 @@ export const SidebarOption = styled.div`
 	border-radius: 6px;
 	transition: all 200ms;
 
+	${({ active }) => active && css`{background-color: ${lighten(0.4, $grey)}`}
```
Comment (arielbk) [inline client/src/components/Dashboard/Dashboard.styled.js]: Is the CSS helper needed? Couldn't this just be:
```javascript
${({ active }) => active && `background-color: ${lighten(0.4, $grey)}`};
```

File: client/src/reducers/habitReducer.js [hunk 2]
```diff
@@ -2,42 +2,30 @@ import Types from '../actions/types';
 
 const { UPDATE_HABIT_LIST, ADD_HABIT, DELETE_HABIT, UPDATE_HABIT } = Types;
 
-const initialState = {
-	habits: [],
```
Comment (arielbk) [inline client/src/reducers/habitReducer.js]: Nice, this needed to be changed 😅

File: client/src/components/Dashboard/Dashboard.styled.js (hunk 1 above)
Comment (Puritanic) [reply client/src/components/Dashboard/Dashboard.styled.js]: Nice catch, I've actually had more properties there but in the end only `background-color` remained. I'll fix this in next PR.

Comment (arielbk) [reply client/src/components/Dashboard/Dashboard.styled.js]: I actually made a small commit to remove it before merging

**Pull request merged**